easybuild/tools/filetools.py

```diff
@@ -1041,8 +1041,13 @@ def apply_regex_substitutions(path, regex_subs, backup='.orig.eb'):
             backup_ext = ''
 
         try:
+            counter = 0
             for line in fileinput.input(path, inplace=1, backup=backup_ext):
```

Comment:
You can use `enumerate` here rather than keeping track of a `counter` yourself (and I would name it `line_id` instead):

```python
for line_id, line in enumerate(fileinput.input(path, inplace=1, backup=backup_ext)):